editor/src/messages/portfolio/document/node_graph/node_graph_message_handler/document_node_types.rs

```diff
@@ -663,15 +729,43 @@ impl DocumentNodeType {
 	}
 }
 
+pub fn wrap_network_in_scope(network: NodeNetwork) -> NodeNetwork {
+	assert_eq!(network.inputs.len(), 1, "Networks wrapped in scope must have exactly one input");
+	let input_type = network.nodes[&network.inputs[0]].inputs.iter().find(|&i| matches!(i, NodeInput::Network(_))).unwrap().clone();
```

Comment:
Perhaps we could do something in the case of multiple network inputs (either adding multiple scopes or adding an error) to avoid strange issues?

Reply:
I think this should basically never happen because a (top level) network can only really have one input (it is basically a function) . I think what we will end up doing is to just not provide the ui to be able to create a network with multiple inputs

Reply:
well, it could be used else where but only in places with a single input. If we were to allow multiple scopes side by side it would be possible to access not yet initialized data. Until we have some sort of borrow checker that is just a limitation of the system

Reply:
Sounds good.

Reply:
![image](https://user-images.githubusercontent.com/21245806/222483150-ac224125-153f-428e-b344-aedafc2fc782.png)